implementations/micrometer-registry-wavefront/src/main/java/io/micrometer/wavefront/WavefrontMeterRegistry.java

```diff
@@ -94,37 +131,54 @@ public void start(ThreadFactory threadFactory) {
         super.start(threadFactory);
     }
 
+    @Override
+    protected DistributionSummary newDistributionSummary(
+        Meter.Id id, DistributionStatisticConfig distributionStatisticConfig, double scale) {
+        if (isWavefrontHistogram(id)) {
```

Comment:
If wavefront histograms collect buckets, then here you are only making a decision about whether to use a wavefront histogram or a plainer distribution summary on the basis of whether `distributionStatisticConfig` has percentile histograms or SLA buckets enabled.

Reply:
Wavefront histograms are reported differently from summaries and other metrics -- they're reported as distributions instead of points.  So deciding whether to use a wfhistogram based on `distributionStatisticConfig` could break backwards compatibility for existing users.  Hence why `isWavefrontHistogram()` is based on the presence of an injected tag.

Reply:
I don't see this as a breaking change. It's likely that a Micrometer `DistributionSummary` can result in _both_ a Wavefront histogram and some other summary statistics based on whether or not `percentileHistograms`/`sla` are configured. This doesn't need to be an either/or proposition. See our treatment of Prometheus timers for example. Prometheus timers don't support the concept of max, so Micrometer ships both a full Prometheus timer and also a separate max gauge.